### ISSUE TITLE
Predict black images

### ISSUE THREAD

**Issue by albertoursino** [guglielmocamporese/hands-segmentation-pytorch#5]
Hello, why the model predicts a black image?
![frame_0011 jpg](https://user-images.githubusercontent.com/56396708/177179016-c4fbe6b7-13c3-4157-bb40-45f888fe2b09.png)
If I pass as input a jpg image:
![frame_0011](https://user-images.githubusercontent.com/56396708/177179043-bb7577c4-5834-4f27-ab38-088d99fb0011.jpg)

Same behaviour with other images.

I downloaded the pretrained model using these lines of code you provided in the README:
```
# Imports
import torch
import torch.hub

# Create the model
model = torch.hub.load(
    repo_or_dir='guglielmocamporese/hands-segmentation-pytorch',
    model='hand_segmentor',
    pretrained=True
)
```

Then I created the dir "checkpoint" on your project directory and put the model there.
Finally, I executed the main passing the predict parameters:
```
python main.py \
	--mode predict \
	--data_base_path './test_images' \
	--model_checkpoint "./checkpoint/checkpoint.ckpt" \
	--model_pretrained
```

Note that I'm using Windows, that's why I didn't run the sh(s) you provided.

Thanks for your time!

**Comment by guglielmocamporese** [comment]
Hi `albertoursino`,

thank you for pointing out this issue. I solved it with the latest commit - this is what I obtained with the fixed code (I used the same commands you reported above in your comment to obtain this output):
![prediction](https://user-images.githubusercontent.com/31989563/177952483-503d971f-2bb4-49b6-bc11-5699ffd2ce3c.gif)


Let me know,

Guglielmo

**Comment by tanveer-hussain** [comment]
def preprocess_image(img):
    transform = T.Compose([T.Resize((256, 256)), T.ToTensor()])
    x = transform(img)
    x = torch.unsqueeze(x, 0)
    x = x.cuda()
    return x

def postprocess_image(preds, H, W):
    # preds = F.softmax(output, 1).argmax(1)[0] * 255 # [h, w]
    preds = Image.fromarray(preds.numpy().astype(np.uint8), 'L')
    preds = preds.resize((W, H))

    return preds

**Comment by guglielmocamporese** [comment]
Hi Tanveer,

Thanks for reaching out - probably the issue is that you are not
normalizing the input correctly.
Here you can find the correct function used as "transform":

transform = transforms.Compose([
>     transforms.Resize((256, 256)),
>     transforms.ToTensor(),
>     lambda x: x if x.shape[0] == 3 else x.repeat(3, 1, 1),
>     transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224,
> 0.225]),

])


Hope this help.
Best,
Guglielmo
